### ISSUE TITLE
Discuss IDEs

### ISSUE THREAD

**Issue by milleniumbug** [LoungeCPP/loungesome-cpp#12]
Which IDEs will go to the awesome list?

**Comment by nabijaczleweli** [comment]
None of them

**Comment by ghost** [comment]
I think VS _is_ awesome, but there's a lot of VC hate around here to just count this as a vote.

Eclipse sucks

Lots of people are really happy with an elaborate vim/ST setup. Probably gotta ask them.

**Comment by nabijaczleweli** [comment]
Neither VIM nor ST3 are IDEs, so they don't count, despite being dope for C++ dev.

@kbok does VisualClang++ work yet?

**Comment by ghost** [comment]
nah

**Comment by nabijaczleweli** [comment]
Until then :-1: for VS

**Comment by Morwenn** [comment]
Code::Blocks is pretty decent but far from being awesome. I liked Netbean's syntax highlighting but that's pretty much the only thing I liked about it.

**Comment by nabijaczleweli** [comment]
Each time I've heard about C::B was in negative contexts, so :-1: from me on that.

**Comment by no-longer-on-githu-b** [comment]
Vim

**Comment by nabijaczleweli** [comment]
Vim is not an IDE.

**Comment by ghost** [comment]
:+1: for vim. When you consider the integration of git, autocompletion, build system, docs, it's foolish to say that it's not an IDE.

**Comment by nabijaczleweli** [comment]
It's a text editor, not an IDE. Vim doesn't have autocompletion (nor many other things) in and of itself.

**Comment by ghost** [comment]
What's itself? Nobody uses vanilla vim. Nobody uses vanilla eclipse platform either, and then argues that it's not an IDE because it doesn't have the autocompletion plugin _if you consider the vanilla version that doesn't have it_. IDEs are modular, so a given configuration of vim which does everything an IDE does, is an IDE.

**Comment by milleniumbug** [comment]
Should we put vim on the awesome list, let's list the plugins we consider awesome together with it. 

IOW let the entry be

Vim + plugin1 + plugin2 + plugin3

instead of

Vim

the same way you wouldn't list (for example) Eclipse, only Eclipse CDT.

**Comment by Morwenn** [comment]
Agreed: if the default plagins are not enough, let's list the plugins required to make some IDE/text editor awesome.

**Comment by no-longer-on-githu-b** [comment]
POSIX is my favourite IDE.

**Comment by ghost** [comment]
@rightfold instead of rightfolding, can you tell us about the plugins you use for C++? Also summoning @sehe, since he will probably also have a contribution to make.

**Comment by nabijaczleweli** [comment]
@rightfold POSIX is a set of documents, and as such it cannot be an IDE

**Comment by Rerito** [comment]
Alrighty then, for Vim I like the following plugins: vim-fugitive (git integration), YouCompleteMe (autocompletion, dope with C++) - requires Clang -, NERDTree (embedded file explorer), gundo (operations history tracking), zeavim (zeal integration for offline docs)

**Comment by no-longer-on-githu-b** [comment]
Pizza Vim-fungitive.

**Comment by nabijaczleweli** [comment]
Bestow us your knowledge, @sehe

**Comment by ThePhD** [comment]
QtCreator is okay, but my only experience with it has been inside of a VM. It also frustrated me because it would throw out my configurations for debug/run when I switched the computers which the project file would open on, without fail. That was always fun...

**Comment by sehe** [comment]
Vim is not an IDE. But it's better at all the tasks IDEs commonly perform for C++.

Clion could have an edge for the future (integrated "visual" debugging and actual refactorings?)

**Comment by nabijaczleweli** [comment]
> Vim is not an IDE.

That we've determined already.

> it's better [...] [than] IDEs 

[What plugins _make_ it better](https://github.com/LoungeCPP/loungesome-cpp/issues/12#issuecomment-176183787), @sehe?

**Comment by sehe** [comment]
Vim is already better.

The plugins make it even better.

Here's my list roughly (it varies)

```
Align
gist-vim
gv.vim
jellybeans.vim
lilypond
nerdcommenter
taglist
textobj-word-column.vim
tinyurl.vim
vim-airline
vim-characterize
vim-commentary
vim-dispatch
vim-fugitive
vim-markdown
vim-misc
vim-monokai
vim-obsession
vim-repeat
vim-sensible
vim-surround
vim-tbone
vim-unimpaired
vim-vinegar
YouCompleteMe

```

**Comment by sehe** [comment]
Frankly the only one that really matters in terms of IDE-fähigkeiten is YouCompleteMe
